pkg/lib/ownerutil/util.go

```diff
@@ -45,6 +45,17 @@ func IsOwnedBy(object metav1.Object, owner Owner) bool {
 	return false
 }
 
+func HasOwnerReference(object metav1.Object, owner metav1.OwnerReference) bool {
+	for _, oref := range object.GetOwnerReferences() {
+		if oref.UID == owner.UID &&
```

Comment:
would just UID be enough?

Reply:
UID might be enough but I don't see why we shouldn't check multiple fields.

Reply:
Removed this function in the latest version of the PR, resolving this conversation as it is no longer relevant.